plugins/modules/ios_logging_global.py

```diff
@@ -0,0 +1,1060 @@
+
+# Before state:
+# -------------
+
+# router-ios#show running-config | section logging
+# logging exception 4099
+# logging message-counter log
+# logging userinfo
+# logging buffered 6000 critical
+# no logging reload
+# no logging rate-limit
+# logging console xml critical
+# logging monitor warnings
+# logging cns-events warnings
+# logging policy-firewall rate-limit 10
+# logging dmvpn rate-limit 10
+# logging trap errors
+# logging facility local6
+# logging host 172.16.1.13 transport tcp port 514
+# logging host 172.16.1.12
+# logging host 172.16.1.10 filtered stream 10
+# logging host 172.16.1.25 filtered
+
+- name: Override commands with provided configuration
+  cisco.ios.ios_logging_global:
+    config:
+      - buffered:
+          severity: critical
+          size: 6000
+      - facility: local6
+      - hosts:
+          - hostname: 172.16.1.25
+            filtered: True
+    state: overridden
+
+# Commands Fired:
+# ---------------
+# "commands": [
+#         "no logging message-counter log",
+#         "no logging host 172.16.1.12",
+#         "no logging host 172.16.1.10",
+#         "no logging host 172.16.1.13",
+#         "no logging exception 4099",
+#         "no logging userinfo",
+#         "no logging console xml critical",
+#         "no logging monitor warnings",
+#         "no logging cns-events warnings",
+#         "no logging policy-firewall rate-limit 10",
+#         "no logging dmvpn rate-limit 10",
+#         "no logging trap errors"
+#     ],
+
+# After state:
+# ------------
+
+# router-ios#show running-config | section logging
+# no logging exception
+# logging buffered 6000 critical
+# no logging reload
+# no logging rate-limit
+# no logging console
+# no logging monitor
+# no logging cns-events
+# no logging trap
+# logging facility local6
+# logging host 172.16.1.25 filtered
+
+# Using state: replaced
+
+# Before state:
+# -------------
+
+# router-ios#show running-config | section logging
+# logging exception 4099
+# logging message-counter log
+# logging userinfo
+# logging buffered xml 5099 notifications
+# no logging reload
+# no logging rate-limit
+# logging console xml critical
+# logging monitor warnings
+# logging cns-events warnings
+# logging policy-firewall rate-limit 10
+# logging dmvpn rate-limit 10
+# logging trap errors
+# logging facility local5
+# logging snmp-trap errors
+# logging host 172.16.1.13 transport tcp port 514
+# logging host 172.16.1.11 xml
+# logging host 172.16.1.12
+# logging host 172.16.1.10 filtered stream 10
+
+- name: Replace commands with provided configuration
+  cisco.ios.ios_logging_global:
+    config:
+      - buffered:
+          severity: alerts
+          size: 6025
+      - facility: local6
+      - hosts:
+          - hostname: 172.16.1.19
+          - hostname: 172.16.1.10
+            filtered: true
+            stream: 15
+    state: replaced
+
+# Commands Fired:
+# ---------------
+
+# "commands": [
+#         "no logging host 172.16.1.13",
```

Comment:
@KB-perByte like we discussed with this, what's the difference b/w `overridden` and `replace` state, ideally imo for list of dict params `replace` should replace content of only 1 config param unlike `overridden` state which will effect all of the params of list of dict, like for config scenario:
```
logging persistent threshold 2
logging facility local6
logging host 172.16.0.1 filtered session-id string test
logging host 10.55.240.143
logging host 10.0.2.15 transport tcp audit
logging host 172.16.0.1 transport tcp
logging host 172.16.0.2
```
Play:
```
 name: Replace commands with provided configuration
  cisco.ios.ios_logging_global:
    config:
      - hosts:
          - hostname: 172.16.0.1
             sequence-num-session: true
```
Output for replace should be:
```
[
  "no logging host 172.16.0.1 filtered session-id string test",
  "no logging host 172.16.0.1 transport tcp",
  "logging host 172.16.0.1 sequence-num-session"
]
```
Output for override:
```
[
  "no logging host 10.55.240.143",
  "no logging host 172.16.0.2",
  "no logging host 172.16.0.1 filtered session-id string test",
  "no logging host 172.16.0.1 transport tcp",
  "logging host 172.16.0.1 sequence-num-session"
]
```

Reply:
As per the discussion we had in the calls/ demo 
**Overridden** - to override the whole logging configuration and put in what is there in the playbook and negating everything else.
Whereas,
**Replaced** - to replace the whole list of **have** configuration for host / other list params individually.
**Tagging in examples-** 
```
Existing running-config -
logging persistent threshold 2
logging facility local6
logging host 172.16.0.1 filtered session-id string test
logging host 10.55.240.143
logging host 10.0.2.15 transport tcp audit
logging host 172.16.0.1 transport tcp
logging host 172.16.0.2
```
Playbook - 
```
  - name: Replace commands with provided configuration
    cisco.ios.ios_logging_global:
      config:
        - hosts: 
            - hostname: 10.0.1.19
      state: replaced
```
Commands fired -
```
no logging host 172.16.0.1
no logging host 10.55.240.143
no logging host 10.0.2.15
no logging host 172.16.0.1
no logging host 172.16.0.2
logging host 10.0.1.19
```
for a case to update a single logging host one can use the merged state.

Reply:
hmm.. I might have missed the discussion then, and if it was discussed and agreed upon then it's OK.